fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ModDiscoverer.java

```diff
@@ -85,8 +87,17 @@ public ModValidator discoverMods() {
                 if (!badModFiles.isEmpty()) {
                     LOGGER.error(LogMarkers.SCAN, "Locator {} returned {} files which is are not ModFile instances! They will be skipped!", locator, badModFiles.size());
                     brokenFiles.addAll(badModFiles.stream().map(IModFile::getModFileInfo).toList());
+                    locatedFiles.removeAll(badModFiles);
                 }
-                locatedFiles.removeAll(badModFiles);
+
+                if (dist.isDedicatedServer()) {
+                    var clientOnlyModFiles = locatedFiles.stream().filter(file -> file.getModFileInfo().isClientSideOnly()).toList();
+                    if (!clientOnlyModFiles.isEmpty()) {
+                        LOGGER.debug(LogMarkers.SCAN, "Locator {} returned {} files which are client-side-only mods, but we're on a dedicated server. They will be skipped!", locator, clientOnlyModFiles.size());
```

Comment:
The dream is users shouldn't have to care, with it still being logged in the debug.log if they're a dev trying to figure out why their mod isn't loading.

But changing it to warn so that people remove it from server zips would make them smaller and start up a bit faster, so meh I'll change it.